Fix `mach test-wpt` to make crash tests work

mukilan opened the pull request:
The first commit fixes two issues related to crash tests:
1. test-wpt is unable to find existing crash tests even when called with --test-types=crashtests. The fix here is to add crashtests to the default test suite types to python/wpt/run.py
2. When running in headless mode, crashes in style threads don't cause servo to crash because the logic in constellation.rs currently calls handle_panic only when the top-level browsing context id is some value. Since style pool threads are shared, they always generate Panic messages with None as top-level browsing context id.

The second commit addresses issues with generating and capturing backtrace and stderr for crash tests.

<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes 

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mukilan commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit bd94242e723fec5ab8c1f3ebace66185ff69fd2e with merge 08f2d9938c77b20ca3848c570b45534162d4b40a...

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/actions/runs/5155522379/jobs/9288478428)

mrobinson commented:
Hrm. It seems like this is causing more timeouts. I wonder what could be preventing the process from exiting in these tests...

mukilan committed:
Fix `mach test-wpt` to make crash tests work

There are two issues related to crash tests:
1. test-wpt is unable to find existing crash tests even when
   called with --test-types=crashtests. The fix here is to
   add crashtests to the default test suite types to python/wpt/run.py
2. When running in headless mode, crashes in style threads
   don't cause servo to crash because the logic in constellation.rs
   currently calls handle_panic only when the top-level browsing
   context id is some value. Since style pool threads are shared,
   they always generate Panic messages with None as top-level
   browsing context id.

Co-authored-by: Martin Robinson <mrobinson@igalia.com>
Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mukilan committed:
Send bactrace to stderr and capture it in test runner

Servo's panic hook writes backtraces to stdout. This
patch changes it so they are written to stderr.

The crash test executor for servo in WPT grouping formatter
was also not capturing the output correctly for crashtests
as the log events were being aggregated based on thread name
which doesn't seem to match correctly in case of crashtests.
This patch also fixes the log grouping logic to be based on
test name.

Co-authored-by: Martin Robinson <mrobinson@igalia.com>
Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mukilan committed:
crashtests: update expectations for layout 2020

Co-authored-by: Martin Robinson <mrobinson@igalia.com>
Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mukilan committed:
crashtests: update expectations for layout 2013

Co-authored-by: Martin Robinson <mrobinson@igalia.com>
Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

servo-wpt-sync commented:
🤖 Opened new upstream WPT pull request (web-platform-tests/wpt#41513) with upstreamable changes.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.

servo-wpt-sync commented:
📝 Transplanted new upstreamable changes to existing upstream WPT pull request (web-platform-tests/wpt#41513).

mrobinson commented:
Looks like the two failures are:

   - PASS [expected CRASH] /custom-elements/when-defined-reentry-crash.html (2013 / 2020)
   - PASS [expected CRASH] /streams/readable-streams/crashtests/strategy-worker-terminate.html (2020)

The first was recently fixed, so I think we can remove that expectation. The second must be a flaky crash. For that I guess we simply need to remove the expectation and open an I-Intermittent bug to track it.

mukilan committed:
remove outdated & intemittent test expectations

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mukilan commented:
Thanks @mrobinson! I've removed the expectations for those tests and repurposed #30124 to track the intermittent CRASH

servo-wpt-sync commented:
📝 Transplanted new upstreamable changes to existing upstream WPT pull request (web-platform-tests/wpt#41513).

mrobinson commented:
I suspect that this failure in the float unit tests is unrelated to this change. The float unit tests use quickcheck to essentially "fuzz" the float layout algorithm. Sometimes the values passed stretch the limits of our floating point layout in Layout 2020 (see #29819). To confirm, I'm going to send this back to the MQ to see if this behavior is flaky.
